extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProviderPlugin.java

```diff
@@ -32,7 +33,7 @@ public SeekableChannelsProvider createProvider(@NotNull final URI uri, @Nullable
             }
             throw new IllegalArgumentException("Arguments not compatible, provided uri " + uri);
         }
-        final S3Instructions s3Instructions = (S3Instructions) config;
+        final S3Instructions s3Instructions = (S3Instructions) Require.neqNull(config, "config");
```

Comment:
I think L31-33 should be *outside* of the `!isCompatible` test. That would be more correct, and would also render this `Require.neqNull` unnecessary.